browser/src/Plugins/Api/Process.ts

```diff
@@ -1,27 +1,46 @@
 import * as ChildProcess from "child_process"
-
 import * as Oni from "oni-api"
+
 import * as Log from "oni-core-logging"
 
 import * as Platform from "./../../Platform"
 import { configuration } from "./../../Services/Configuration"
 
 export interface IShellEnvironmentFetcher {
-    getEnvironmentVariables(): Promise<any>
+    getEnvironmentVariables(): Promise<NodeJS.ProcessEnv>
 }
 
 export class ShellEnvironmentFetcher implements IShellEnvironmentFetcher {
-    private _shellEnvPromise: Promise<any>
-    private _shellEnv: any
+    public _env: NodeJS.ProcessEnv
+
+    constructor(private _shellEnvPromise = import("shell-env")) {}
+
+    public async getEnvironmentVariables(): Promise<NodeJS.ProcessEnv> {
+        if (!this._env) {
+            const shellEnv = await this._shellEnvPromise
+            // TODO:
+            // Shell Env currently doesn't derive the users
+            // shell correctly for non-Windows systems
+            // https://github.com/sindresorhus/default-shell/issues/3
+            // const { shell } = os.userInfo() - this accomplishes that
+            // though issue here is that it reads the relevant dotfile which
+            // if it has issues will stop oni from starting up
+            try {
+                this._env = shellEnv.default.sync()
+            } catch {
+                this._env = process.env
+            }
+            this._fixPath()
+        }
 
-    constructor() {
-        this._shellEnvPromise = import("shell-env")
+        return this._env
     }
 
-    public async getEnvironmentVariables(): Promise<any> {
-        if (!this._shellEnv) {
-            this._shellEnv = await this._shellEnvPromise
-            return this._shellEnv.sync()
+    private _fixPath() {
+        // Set the PATH to a derived path on MacOS as this is not set correctly in electron
+        // the idea is based on https://github.com/sindresorhus/fix-path
+        if (Platform.isMac() && this._env.PATH) {
+            process.env.PATH = this._env.PATH
```

Comment:
Hmm, I don't think it makes sense for us to ever change `process.env.PATH` ourselves. From my understanding, all the cases where we need the full `path` are in _spawned processes_, and in those cases, we can send this 'augmented' path to them - which is what `mergeSpawnOptions` does today. 

To me, it seems like the root problem is that, in some cases, we aren't sending the `PATH` properly to the spawned processes. Setting `process.env.PATH` directly here is a janky workaround that is really risky and susceptible to heisenbugs, so I'd rather us understand the root cause and fix it.

In other words - why would we need to do this if Node gives us the ability to send `PATH` to spawned processes? It makes sense that we need to get the proper shell path, but for things like #2251, it should be sufficient to send it to the `nvim` binary. If it's not making it there, we should understand why that pipeline isn't working.

Reply:
@bryphe happy to leave this be and leave this change as a simple rectification of the the import error (which is unfortunately definitely failing in master), and remove the above changes.